spec/index.html

```diff
@@ -857,30 +869,227 @@ <h3>Simple N3 interpretations</h3>
+			<li><b>inputs and outputs</b>: <a>input arguments</a> and <a>output arguments</a>.</li>
+			<li><b>summary</b>: a human-readable summary of the function of the built-in.</li>
+		</ul>
+		
+		We define these concepts in more detail below.
+	</p>
+	
+	<section id="builtins_args">
+		<h4>Input and output arguments</h4>
+		<p>An N3 <a>built-in</a> operates on its arguments. An <dfn>argument</dfn> is a placeholder that refers to an <a>N3 triple element</a> from the <dfn>built-in statement</dfn>, i.e., the <a>N3 statement</a> where the built-in acts as a predicate.</p>
+		<p>In the simplest case, there are two arguments that respectively refer to the subject and object of the built-in statement. For instance, statements such as `1 math:lessThan 2` are true if the subject is less than the object. Hence, built-in `math:lessThan` has two arguments `$a_1` and `$a_2`, which is written as follows: `$a_1 math:lessThan $a_2`. (In this section, arguments are prefixed with `$a`.)</p>
+		<p>Arguments can be also represent a "deconstruction" of the subject or object in case of lists or cited formulas. For instance, `(1 2 3) math:sum 6` calculates the sum of a list of numbers. Hence, the built-in `math:sum` has arguments `$a_1` .. `$a_n` and `$a_sum`, which is written as follows: `($a_1 .. $a_n) math:sum $a_sum`. </p>			
+		<p>In a built-in statement, if an argument refers to a concrete <a>N3 triple element</a> (i.e., either a concrete literal or a bound variable) within the built-in statement, we call its referent a built-in <dfn>input</dfn>, else, we call its referent a built-in <dfn>output</dfn>.
+		<p>In some cases, an N3 built-in will only support outputs for one argument:
+			e.g., `math:sum` only supports outputs for `$a_sum`.
+			In other cases, an N3 built-in only supports inputs for all arguments: 
+			e.g., `math:lessThan` only supports inputs for `$a_1` and `$a_2`.
+			In rarer cases, an N3 built-in can support outputs for several arguments:
+			e.g., for `35 math:cos ?x`, the `math:cos` of `35` will be calculated; for `?x math:cos 0.57` the inverse operation (`math:acos`) for `0.57` will be calculated.
+		</p>
+		<p>The set of arguments, and whether they are allowed to have inputs or outputs, are defined per N3 built-in. An argument that is allowed to have output referents is called an <dfn>output argument</dfn>; otherwise, it is called an <dfn>input argument</dfn>.</p>
+		<p class="ednote">The choice of allowing an argument as output is typically dictated by the need to avoid an infinite amount of results, intractable calculations, and general constraints that weigh utility vs. implementation difficulty.<br /><br />
+		For instance, it would be perfectly possible to allow a single operand argument in `math:sum` to be an output in case the result argument is an input (e.g., `(1 ?o ?3) math:sum 6`) &mdash; but this complicates the implementation of the built-in while offering only limited utility.</p>
```

Comment:
```suggestion
		For instance, it would be perfectly possible to allow a single operand argument in `math:sum` to be an output in case the result argument is an input (e.g., `(1 ?o 3) math:sum 6`) &mdash; but this complicates the implementation of the built-in while offering only limited utility.</p>
```